stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -274,15 +278,20 @@ public class MessageListViewModel(
     }
 
     /**
-     * Used to filter messages deleted by other users.
+     * Used to filter messages which we should show to the current user.
      *
      * @param messages List of all messages.
      * @return Filtered messages.
      */
-    private fun filterDeletedMessages(messages: List<Message>): List<Message> {
+    private fun filterMessagesToShow(messages: List<Message>): List<Message> {
         val currentUser = user.value
 
-        return messages.filter { !(it.user.id != currentUser?.id && it.deletedAt != null) }
+        return messages.filter {
+            val isNotDeletedByOtherUser = !(it.user.id != currentUser?.id && it.deletedAt != null)
```

Comment:
Possibly just a small optimization, but would it not be prudent to write this condition as:

```kotlin
!(it.deletedAt != null && it.user.id != currentUser?.id &&)
```
The reasoning behind this is as follows:

Fewer messages are deleted than are owned by others.  Logical operator `&&` will make sure that whatever is to the right of it does not get evaluated if whatever is to the left of it is `false` since it already knows the condition cannot be satisfied.

This would save unnecessary evaluation.

Reply:
That's actually really good - if the message is not deleted, we save 1 comparison. 

Nicely done! :]